### ISSUE TITLE
Test suite fails (with newest Test::More and/or Test2::*?)

### ISSUE THREAD

**Issue by eserte** [uperl/Test2-Tools-JSON-Pointer#1]
See subject. Failing test log:
```
    # Failed test 'hash fail sans pointer'
    # at t/test2_tools_json_pointer.t line 57.
    # +---------------+-----+---------------+----+---------------+--------+
    # | PATH          | LNs | GOT           | OP | CHECK         | LNs    |
    # +---------------+-----+---------------+----+---------------+--------+
    # |               |     | ARRAY(0x41595 |    | <ARRAY>       | 41, 56 |
    # |               |     | a68)          |    |               |        |
    # |               |     |               |    |               |        |
    # | [0]           | 38  | Test2::Event: |    | <EVENT: Ok>   | 42     |
    # |               |     | :Fail=HASH(0x |    |               |        |
    # |               |     | 41f53978)     |    |               |        |
    # |               |     |               |    |               |        |
    # | [0] <blessed> |     | Test2::Event: | eq | Test2::Event: |        |
    # |               |     | :Fail         |    | :Ok           |        |
    # |               |     |               |    |               |        |
    # | [0]{pass}     |     | <DOES NOT EXI |    | 0             |        |
    # |               |     | ST>           |    |               |        |
    # |               |     |               |    |               |        |
    # | [1]           |     | <DOES NOT EXI |    | <EVENT: Diag> | 43     |
    # |               |     | ST>           |    |               |        |
    # |               |     |               |    |               |        |
    # | [2]           |     | <DOES NOT EXI |    | <EVENT: Diag> | 47, 47 |
    # |               |     | ST>           |    |               |        |
    # +---------------+-----+---------------+----+---------------+--------+
    # Failed test 'hash fail with pointer'
    # at t/test2_tools_json_pointer.t line 81.
    # +---------------+-----+---------------+----+---------------+--------+
    # | PATH          | LNs | GOT           | OP | CHECK         | LNs    |
    # +---------------+-----+---------------+----+---------------+--------+
    # |               |     | ARRAY(0x41f4e |    | <ARRAY>       | 65, 80 |
    # |               |     | 9f0)          |    |               |        |
    # |               |     |               |    |               |        |
    # | [0]           | 62  | Test2::Event: |    | <EVENT: Ok>   | 66     |
    # |               |     | :Fail=HASH(0x |    |               |        |
    # |               |     | 41735af8)     |    |               |        |
    # |               |     |               |    |               |        |
    # | [0] <blessed> |     | Test2::Event: | eq | Test2::Event: |        |
    # |               |     | :Fail         |    | :Ok           |        |
    # |               |     |               |    |               |        |
    # | [0]{pass}     |     | <DOES NOT EXI |    | 0             |        |
    # |               |     | ST>           |    |               |        |
    # |               |     |               |    |               |        |
    # | [1]           |     | <DOES NOT EXI |    | <EVENT: Diag> | 67     |
    # |               |     | ST>           |    |               |        |
    # |               |     |               |    |               |        |
    # | [2]           |     | <DOES NOT EXI |    | <EVENT: Diag> | 71, 71 |
    # |               |     | ST>           |    |               |        |
    # +---------------+-----+---------------+----+---------------+--------+

# Failed test 'JSON comapre fail'
# at t/test2_tools_json_pointer.t line 84.
    # Failed test 'invalid JSON'
    # at t/test2_tools_json_pointer.t line 116.
    # +--------------+-----+--------------+----+--------------+----------+
    # | PATH         | LNs | GOT          | OP | CHECK        | LNs      |
    # +--------------+-----+--------------+----+--------------+----------+
    # |              |     | ARRAY(0x41f5 |    | <ARRAY>      | 100, 115 |
    # |              |     | 3e40)        |    |              |          |
    # |              |     |              |    |              |          |
    # | [0]          | 97  | Test2::Event |    | <EVENT: Ok>  | 101      |
    # |              |     | ::Fail=HASH( |    |              |          |
    # |              |     | 0x41f506d8)  |    |              |          |
    # |              |     |              |    |              |          |
    # | [0] <blessed |     | Test2::Event | eq | Test2::Event |          |
    # | >            |     | ::Fail       |    | ::Ok         |          |
    # |              |     |              |    |              |          |
    # | [0]{pass}    |     | <DOES NOT EX |    | 0            |          |
    # |              |     | IST>         |    |              |          |
    # |              |     |              |    |              |          |
    # | [1]          |     | <DOES NOT EX |    | <EVENT: Diag | 102      |
    # |              |     | IST>         |    | >            |          |
    # |              |     |              |    |              |          |
    # | [2]          |     | <DOES NOT EX |    | <EVENT: Diag | 105      |
    # |              |     | IST>         |    | >            |          |
    # +--------------+-----+--------------+----+--------------+----------+

# Failed test 'invalid json'
# at t/test2_tools_json_pointer.t line 119.
t/test2_tools_json_pointer.t .............. 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/4 subtests 
```

Statistical analysis shows a number of candidates: some belonging to the Test2::* namespace and Test::More (@exodist: FYI).

**Comment by exodist** [comment]
Test2::Suite caused it, but the fix needs to be made in Test2-Tools-JSON-Pointer. Parts of Test2::Suite now use Pass/Fail events with the diagnostics directly linked into the events instead of separate ok/diag events.

I am also working on a way to make tests that verify events less fragile so that we can avoid issues like this in the future.

**Comment by plicease** [comment]
Thanks for the report and the analysis.  Should be fixed in 0.02.
